src/lib.rs

```diff
@@ -653,6 +637,11 @@ impl Context {
             }
             let rs = RsPath::new(p.to_path_buf());
             try!(warn_if_nonexistant(&rs));
+
+            if self.emit_rerun_if {
+                println!("cargo:rerun-if-changed={}", &rs.display());
```

Comment:
this is what I expected to see.

I think you will need a similar bit of code in the loop below here that is processing the `.md` input files.